util/src/main/java/tech/pegasys/teku/util/async/SafeFuture.java

```diff
@@ -87,6 +88,47 @@ public static void reportExceptions(final CompletionStage<?> future) {
     }
   }
 
+  /**
+   * Creates a completed {@link SafeFuture} instance if no of supplied interruptors is completed,
+   * else creates an exceptionally completed {@link SafeFuture} instance
+   *
+   * @see #orInterrupt(Interruptor...)
+   */
+  public static SafeFuture<Void> notInterrupted(Interruptor... interruptors) {
+    return SafeFuture.<Void>completedFuture(null).orInterrupt(interruptors);
+  }
+
+  /**
+   * Create {@link Interruptor} with default exception supplier which creates {@link
+   * InterruptedException}
+   *
+   * @see #createInterruptor(CompletableFuture, Supplier)
+   */
+  public static Interruptor createInterruptor(CompletableFuture<?> interruptFuture) {
```

Comment:
Doesn't look like we use this - seems reasonable to require a custom exception on interruption ...